docs/user/admin/settings.rst

```diff
@@ -630,3 +630,84 @@ Request ::
         }
       }
     }
+
+plugins.query.datasources.enabled
+=================================
+
+Description
+-----------
+
+This setting controls whether datasources are enabled.
```

Comment:
does this setiting impact _async_query API? if plugins.query.datasources.enabled=false, _async_query will fail?

Reply:
That's a good question. I did not test this case. I will add an integ test to validate the behavior.

Reply:
Update : I've added additional checks in the async query code path to validate that this setting is also enabled.
This PR also contains integ tests for that case as well.